### DISCUSSION TITLE
Cypress 10 and React-Aria HiddenSelect

### DISCUSSION THREAD

**Discussion by dbani-dev** [adobe/react-spectrum#3618] [Q&A] [answered]
While running a custom `react-aria` Select component test in Cypress 10

```
describe('select.cy.ts', () => {
  it('select can render', () => {
    cy.mount(
      <Select>
        <Item key="Item A">Item A</Item>
        <Item key="Item B">Item B</Item>
        <Item key="Item C">Item C</Item>
      </Select>
    );
  });
});
```

I received the following **error (x3)**:

```
TypeError: Cannot read properties of undefined (reading 'type')
at HiddenSelect.tsx:116:24
```

It seems that the useSelectState -> `state.collection.getKeys()` is `undefined`, instead of `MapIterator {'Item A', 'Item B', 'Item C'}`.

Is this a Cypress or React-Aria hidden select issue?

**Comment by snowystinger** [comment, marked as answer]
We're unaware of any issues with our HiddenSelect component. Can you provide a minimal reproduction? That can help us debug it. Otherwise, we're just making guesses.

**Reply by dbani-dev** [reply]
Running the `Select` locally throws no errors. It may be hard to produce an example for component testing. Running in Cypress v10.9.0 (latest) in component testing mode throws type property errors. 

<img width="2084" alt="Screen Shot 2022-10-11 at 9 09 24 am" src="https://user-images.githubusercontent.com/1149860/194975349-78666c37-e8bf-4dab-882c-ae53aac99228.png">

Item in collection is `undefined`:

<img width="1853" alt="Screen Shot 2022-10-11 at 9 11 07 am" src="https://user-images.githubusercontent.com/1149860/194975478-d5592572-9901-4fda-b617-a41b28d8a685.png">

I'm guessing Cypress is handling the `HiddenSelect` in a weird way? If `HiddenSelect` is removed, the `Select` renders but doesn't have `Item`(s).

**Reply by snowystinger** [reply]
It's fine if the reproduction is in Cypress, we just need a github repo with a minimal app (just the Select) and a single cypress test that reproduces it so we can help with debugging. I'd just be guessing right now, and I don't have any good guesses.

**Reply by dbani-dev** [reply]
Issue has been fixed. Apologies, NX was producing incorrect types.
